cloudsync_onedrive.py

```diff
@@ -1327,37 +1342,42 @@ def namespace_id(self, ns_id: str):
-            self._fetch_drive_list()
-            drive = self.__drive_by_id.get(ns_id)
-            if not drive:
-                ids = Drive(ns_id, ns_id)
-                if ids.site_id:
-                    site = self.__site_by_id.get(ids.site_id)
-                    if not site:
-                        raise CloudNamespaceError(f"Unknown site id: {ns_id}")
-                    self._fetch_drives_for_site(site)
-                    drive = self.__drive_by_id.get(ns_id)
-                    if not drive:
-                        # check if it is a shared drive
-                        drive = next((d for d in self._shared_with_me.drives if d.drive_id == ids.drive_id), None)
-                    if not drive:
-                        raise CloudNamespaceError(f"Site does not contain drive: {ns_id}")
-                elif ids.drive_id:
-                    if ids.drive_id == self._personal_drive.drives[0].drive_id:
-                        drive = self._personal_drive.drives[0]
-                    else:
-                        api_drive = self._direct_api("get", f"/drives/{ids.drive_id}/")
-                        if not api_drive:
-                            raise CloudNamespaceError(f"Unknown drive id: {ns_id}")
-                        drive = Drive(api_drive.get("name", "Personal"), ns_id)
-                else:
-                    raise CloudNamespaceError(f"Malformed drive id: {ns_id}")
-            self._namespace = drive
+            self._namespace = self._get_validated_namespace(ns_id)
         else:
             # defer validation until a connection is established
             self._namespace = self.__drive_by_id.get(ns_id, Drive(name=ns_id, id=ns_id))
         log.info("USING NS name=%s id=%s - connected=%s", self.namespace.name, self.namespace_id, self.connected)
 
+    def _get_validated_namespace(self, ns_id: str):
+        drive = self.__drive_by_id.get(ns_id)
+        if not drive:
+            self._fetch_drive_list()
+            drive = self.__drive_by_id.get(ns_id)
+        if not drive:
+            ids = Drive(ns_id, ns_id)
+            if ids.site_id:
+                site = self.__site_by_id.get(ids.site_id)
+                if not site:
+                    raise CloudNamespaceError(f"Unknown site id: {ns_id}")
+                self._fetch_drives_for_site(site)
+                drive = self.__drive_by_id.get(ns_id)
+                if not drive:
+                    # check if it is a shared drive
+                    drive = next((d for d in self._shared_with_me.drives if d.drive_id == ids.drive_id), None)
+                if not drive:
+                    raise CloudNamespaceError(f"Site does not contain drive: {ns_id}")
+            elif ids.drive_id:
+                if ids.drive_id == self._personal_drive.drives[0].drive_id:
+                    drive = self._personal_drive.drives[0]
+                else:
+                    api_drive = self._safe_direct_api(f"/drives/{ids.drive_id}/")
```

Comment:
This will treat an error from the underlying API as an affirmative `Unknown drive id`. Should a default be provided for this call to `_safe_direct_api` so that the condition where "a successful request returned no api_drive" can be distinguished from "a failure to complete the request at all"?

Reply:
reverted this back to the standard `_direct_api()` call